src/measure_theory/probability_mass_function/basic.lean

```diff
@@ -58,6 +58,25 @@ lemma coe_le_one (p : pmf α) (a : α) : p a ≤ 1 :=
 has_sum_le (by { intro b, split_ifs; simp only [h, zero_le'] })
   (has_sum_ite_eq a (p a)) (has_sum_coe_one p)
 
+lemma apply_eq_one_iff (p : pmf α) (a : α) :
+  p a = 1 ↔ p.support = {a} :=
+begin
+  refine ⟨λ h, set.eq_of_subset_of_subset _ _, λ h, le_antisymm (pmf.coe_le_one p a) (le_of_eq _)⟩,
+  { refine λ a' ha', by_contradiction (λ h', false.elim (ne_of_gt _ p.tsum_coe)),
```

Comment:
Do we have `p s = 1 -> p.support \subset s`?

Reply:
There is `to_outer_measure_apply_eq_one_iff`. Making a detour through the measure theory stuff for this proof simplified it a good bit, so I added a few lemmas and proved it that way.